tests/main.cpp

```diff
@@ -0,0 +1,56 @@
+
+#include "FPSciTests.h"
+
+#include <gtest/gtest.h>
+#include <FPSciApp.h>
+#include <crtdbg.h>
+
+int AbortReportHook(int reportType, char* message, int* returnValue)
+{
+	const char* typeStr;
+	switch (reportType) {
+	case _CRT_WARN: typeStr = "Warning"; break;
+	case _CRT_ERROR: typeStr = "Error"; break;
+	case _CRT_ASSERT: typeStr = "Assertion"; break;
+	default: typeStr = "<invalid report type>"; break;
+	}
+	printf("Abort (%s): %s\n", typeStr, message);
+	fflush(stdout);
+	*returnValue = 1;
+	return true; // no popup!
+}
+// Tells C++ to invoke command-line main() function even on OS X and Win32.
+G3D_START_AT_MAIN();
+
+int main(int argc, const char** argv)
+{
+	// Stop visual studio creating an abort popup and stalling the CI runner
+	_CrtSetReportHook(AbortReportHook);
+
+	// Hack to disable error popups during automated testing because there is no dynamic version of G3D_DEBUG_NOGUI
+	G3D::_internal::_consolePrintHook = nullptr;
+	G3D::_internal::_debugHook = nullptr;
+	G3D::_internal::_failureHook = nullptr;
+
+	::testing::InitGoogleTest(&argc, const_cast<char**>(argv));
```

Comment:
VS is giving red underlines here. Is something missing to make this seem to work more correctly to visual studio?

Reply:
Perhaps the const cast? If it compiles I'd claim an intellisense bug and call it a day. Personally I think argv should be const, and googletest could happily take a const anyway, but main typically has a non-const argv.

Reply:
It's complaining about the `::` with nothing in front of them. Shouldn't there normally be a namespace or something on the left hand side of the `::` operator?

Reply:
`::` alone is valid C++. It's an explicit way to refer to the global namespace in case a local/class member has the same name. I doubt it's a new feature either. This line is also from the official googletest docs: https://github.com/google/googletest/blob/master/googletest/docs/primer.md

Nevertheless I've removed it. Hopefully it's happy now?